json-smart/src/main/java/net/minidev/json/parser/JSONParserBase.java

```diff
@@ -39,6 +39,12 @@
  */
 abstract class JSONParserBase {
 	protected char c;
+	/**
+	 * hard coded maximal depth for JSON parsing
+	 */
+	public final static int MAX_DEPTH = 400;
```

Comment:
maybe `MAX_DEPTH` should not be final.
If I merge this PR, somes will ask me to add code to change this value... Opensource is like that...
please, leave it as non final.